### PR TITLE
Use tmp path for MNIST dataset

### PR DESCRIPTION
**Pull request by forbearer**

### CODE REVIEW

File: examples/mnist/README.md [hunk 1]
```diff
@@ -7,18 +7,18 @@ PYTHONPATH=~/dev/petastorm  # replace with your petastorm install path
 
 ## Generating a Petastorm Dataset from MNIST Data
 
-This creates both a `train` and `test` petastorm datasets in `/home/${USER}/dev/datasets/mnist`:
+This creates both a `train` and `test` petastorm datasets in `/tmp/mnist`:
 
 ```bash
-python generate_petastorm_mnist.py -o file:///home/${USER}/dev/datasets/mnist
+python generate_petastorm_mnist.py -o file:///tmp/mnist
```
[the remark this answers is not in the document]
Comment (forbearer) [reply examples/mnist/README.md]: Ah, yes, I should've guessed that you'd want that by this point!  :-)

**Comment by forbearer** [comment]
@selitvin While testing the default, if the dataset was generated once before, the generate will complain and fail.

Originally, I had overwrite set `true`, but removed that at @rgruener's suggestion.

Do you think the erroring out is OK?

**Comment by rgruener** [comment]
I am not so opinionated on it but I do think erroring out is OK. The output of this shouldnt change so if you already created a dataset, I think its fine to tell you and just mention that you have the option to overwrite it. I think the important thing is that it works easily for people using it for the first time, rather than necessarily every time

**Comment by selitvin** [comment]
Agree with @rgruener I am ok with erroring out.

**Pull request merged**